buildSrc/src/main/java/org/elasticsearch/gradle/precommit/ThirdPartyAuditTask.java

```diff
@@ -348,6 +350,11 @@ private String runForbiddenAPIsCli() throws IOException {
             }
             spec.setIgnoreExitValue(true);
         });
+        if (OS.current().equals(OS.LINUX) && result.getExitValue() == SIG_KILL_EXIT_VALUE) {
```

Comment:
Does the process return a stable exit code on a "normal" failure? If so, could we just check to see if the exit code was anything but the expected one and return an error?

Also, I don't think we need to filter this to just Linux. If this process is killed or terminates awkwardly for some reason on any other OS that's worth reporting even if it's unlikely to happen.

Reply:
So it looks as though we can check for [expected codes](https://github.com/policeman-tools/forbidden-apis/blob/master/src/main/java/de/thetaphi/forbiddenapis/cli/CliMain.java#L63) here. I'd have this barf if we get anything but a code of `0` or `1` back. We can still additionally check that if the code is `137` we report the message as you have here, indicating that OOM might be a culprit.

Reply:
I was under the impression that I tried it when implementing this and it didn't work ...
I checked it again and it does work, so I added that check for expected codes now. 
I also added a limit to the heap so we don't confuse the oom killer by vastly over-committing memory 
when running in parallel.